<launch>
    <arg name="downsample_factor" default="4"/>
    <arg name="grid_size_limit" default="50"/>
    <arg name="sensor_data_topic" default="/sensor_data"/>
    <arg name="frontier_points_topic" default="/frontier_points"/>
    <arg name="map_topic" default="/map"/>
    <arg name="abstract_map_topic" default="/abstract_map"/>
    <arg name="agent_state_topic" default="/agent_state"/>
    <arg name="debug_grid_topic" default="/debug_grid"/>
    
    <node name="map_processor" pkg="turtlebot3_explorer" type="map_processor.py" output="screen">
        <param name="downsample_factor" value="$(arg downsample_factor)"/>
        <param name="grid_size_limit" value="$(arg grid_size_limit)"/>
        <remap from="/sensor_data" to="$(arg sensor_data_topic)"/>
        <remap from="/frontier_points" to="$(arg frontier_points_topic)"/>
        <remap from="/map" to="$(arg map_topic)"/>
        <remap from="/abstract_map" to="$(arg abstract_map_topic)"/>
        <remap from="/agent_state" to="$(arg agent_state_topic)"/>
        <remap from="/debug_grid" to="$(arg debug_grid_topic)"/>
    </node>
    
</launch>